### PR TITLE
Minor Parquet cursor fixes

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Seems like the order of the calls to end a ``ParquetJsonColumnConverter`` is not correct.

### CODE REVIEW

**Comment by cberner** [comment]
Can you add a test that reproduces this?

**Comment by nezihyigitbasi** [comment]
@cberner Will think on a simple test case to reproduce this issue. I noticed this with a row, say ``row<f1,f2,f3>``, where ``f1`` is an optional field **and** null (so Parquet doesn't even call the ``addXYZ`` methods in the converters), ``f2`` and ``f3`` are required. Then what happened is we got a corrupt json ``"[f2, f3] null"`` where ``null`` is written after the json array is closed due to the existing logic.

**Comment by nezihyigitbasi** [comment]
@cberner just added a test case and rebased.

**Comment by cberner** [comment]
I think there's a simpler way to test this. Just add this to the `AbstractTestHiveFileFormats`

```
.add(new TestColumn("t_array_nulls", getStandardListObjectInspector(javaDoubleObjectInspector), Arrays.asList(7.0, null) , "[7.0, null]"))
```

However, that test doesn't pass even after this patch is applied, so can you look into whether there's a second bug?

**Comment by nyigitbasi** [comment]
Weird. The current test (my version) passes. Will take a look at why your proposal fails when I find some time. Thanks for the suggestion btw.

**Comment by nezihyigitbasi** [comment]
Weird. The current test (my version) passes. Will take a look at why your proposal fails when I find some time. Thanks for the suggestion btw.

**Comment by nezihyigitbasi** [comment]
@cberner Added two unit tests to reproduce the problem. I also fixed another bug that we hit in production (also reproducible with these test cases).

File: presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java [hunk 1]
```diff
@@ -395,21 +396,24 @@ public PrestoReadSupport(List<HiveColumnHandle> columns, MessageType messageType
...
-                            case MAP_KEY_VALUE: // original versions of Parquet have map and entry swapped
-                                ParquetJsonColumnConverter mapConverter = new ParquetJsonColumnConverter(new ParquetMapJsonConverter(groupType.getName(), null, groupType), i);
-                                converters.add(mapConverter);
-                                closeableBuilder.add(mapConverter);
-                                break;
-                            case UTF8:
-                            case ENUM:
-                                throw new IllegalArgumentException("Group column " + groupType.getName() + " type " + groupType.getOriginalType() + " not supported");
+                        OriginalType originalType = groupType.getOriginalType();
+                        if (originalType == null) { // struct does not have an original type
```
Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java]: This case is handled properly in ``createJsonConverter()`` method, but not here causing NPE.

**Comment by cberner** [comment]
Tried to merge this, but I got a lot of merge conflicts. Could you rebase?

**Comment by cberner** [comment]
Merged. Thanks!

**Comment by nezihyigitbasi** [comment]
@cberner sorry couldn't rebase sooner as I am still on leave. Thanks!

**Comment by cberner** [comment]
Np. Turned out to be pretty simple, when I looked closer at the conflicts :)

**Pull request closed without merging**